src/lib/classes/services/chainsService.ts

```diff
@@ -120,15 +135,62 @@ export class ChainsService {
         ]);
 
         this._cachedChains[chainID].blocks[blockIndex] = block;
-        this.save();
+
+        this.cacheBlocksChain(chainID, blockIndex, block); // Cache the block
 
         return block;
     }
 
-    /**
-     * Persist the cached blocks
-     */
-    public save() {
-        this._storageService.save("chains", this._cachedChains);
+    private cacheBlocksChain(chainID: string, blockIndex: number, block: BlockData) {
```

Comment:
We can rename this as you are caching only 1 block extra, and blockIndex shouldnt be necessary as you already have it in `block`
```suggestion
    private cacheChainBlock(chainID: string, block: BlockData) {
```